scripts/init.groovy.d/mesos-auth.groovy

```diff
@@ -0,0 +1,51 @@
+import com.cloudbees.plugins.credentials.*
+import com.cloudbees.plugins.credentials.domains.*
+import com.cloudbees.plugins.credentials.impl.*
+import hudson.tasks.*
+import jenkins.model.*
+import org.jenkinsci.plugins.mesos.MesosCloud
+
+def changePassword = { userName ->
+  def cloud = MesosCloud.get()
+  def credentialsId = cloud.getCredentialsId()
+  def credId = "mesos-${userName}"
+
+  if (credentialsId && credentialsId == credId) {
+    // do nothing if credential already exists
+    println "--> [mesos] credentials already selected"
+  } else {
+    // create a new credential with an expected ID
+    println "--> [mesos] creating new credentials"
+    String randomPwd = org.apache.commons.lang.RandomStringUtils.random(9, true, true)
+
+    mesosFrameworkCreds = new UsernamePasswordCredentialsImpl(
+      CredentialsScope.GLOBAL,
+      "mesos-${userName}",
+      "mesos authentication",
+      userName, randomPwd)
+    SystemCredentialsProvider.getInstance().getStore().addCredentials(Domain.global(), mesosFrameworkCreds)
+    cloud.setCredentialsId(mesosFrameworkCreds.getId())
+    Jenkins.getInstance().save()
+    cloud.restartMesos()
+
+    println "--> [mesos] creating new credentials... done"
```

Comment:
A dumb nit: "done creating new credentials" instead? The "..." implies work is being done between the start and end of the log output.

Reply:
I copied this from format from an existing init hook:

```
println "--> setting agent port for jnlp"
....
println "--> setting agent port for jnlp... done"
```

Reply:
Ok. +1 to following an existing pattern.